src/data/arena-tag.ts

```diff
@@ -925,53 +925,39 @@ export class DelayedAttackTag extends ArenaTag {
 }
 
 /**
- * Arena Tag class for {@link https://bulbapedia.bulbagarden.net/wiki/Stealth_Rock_(move) Stealth Rock}.
- * Applies up to 1 layer of Stealth Rocks, dealing percentage-based damage to any Pokémon
- * who is summoned into the trap, based on the Rock type's type effectiveness.
+ * Arena Tag class for {@link https://bulbapedia.bulbagarden.net/wiki/Stealth_Rock_(move) Stealth Rock}
+ * and {@link https://bulbapedia.bulbagarden.net/wiki/G-Max_Steelsurge_(move) G-Max Steelsurge}.
+ * Applies up to 1 layer of hazards, dealing percentage-based damage to any Pokemon summoned into the trap
+ * based on the type effectiveness
  */
-class StealthRockTag extends ArenaTrapTag {
-  constructor(sourceId: number, side: ArenaTagSide) {
-    super(ArenaTagType.STEALTH_ROCK, Moves.STEALTH_ROCK, sourceId, side, 1);
+class TypeHazardTag extends ArenaTrapTag {
+  public damagingType: Type;
+
+  constructor(arenaTagType: ArenaTagType, sourceMove: Moves, sourceId: number, side: ArenaTagSide, damagingType: Type) {
+    super(arenaTagType, sourceMove, sourceId, side, 1);
+    this.damagingType = damagingType;
   }
 
   override onAdd(arena: Arena, quiet: boolean = false): void {
     super.onAdd(arena);
 
     const source = this.sourceId ? globalScene.getPokemonById(this.sourceId) : null;
     if (!quiet && source) {
-      globalScene.queueMessage(
-        i18next.t("arenaTag:stealthRockOnAdd", { opponentDesc: source.getOpponentDescriptor() }),
-      );
+      if (this.damagingType === Type.ROCK) {
+        globalScene.queueMessage(
+          i18next.t("arenaTag:stealthRockOnAdd", { opponentDesc: source.getOpponentDescriptor() }),
+        );
+      } else {
+        globalScene.queueMessage(
+          i18next.t("arenaTag:sharpSteelOnAdd", { opponentDesc: source.getOpponentDescriptor() }),
+        );
+      }
```

Comment:
You probably want to avoid hard-coding those messages and instead provide the message as parameter.